src/MOI_wrapper/nonlinear_constraints.jl

```diff
@@ -0,0 +1,124 @@
+# Nonlinear constraints (objective not supported)
+
+MOI.supports(::Optimizer, ::MOI.NLPBlock) = true
+
+function MOI.set(o::Optimizer, ::MOI.NLPBlock, data::MOI.NLPBlockData)
+    # We don't actually store the data (to be queried later during the solve
+    # process). Instead, we extract the expression graphs and add the
+    # corresponding constraints to the model directly.
+    if data.has_objective
+        @warn "Nonlinear objective not supported by SCIP.jl!" maxlog=1
```

Comment:
Shouldn't this be an error instead of a warning if the solver isn't capable of solving the problem that the user wrote down?

Reply:
True. But there is also no way for the solver to signal that it can't handle them, because the `supports` method is just about `NLPBlock` (all or nothing).

Reply:
`error()` is a valid way to signal this :). I think that's better than returning a solution that is almost surely incorrect. Warnings are helpful but it's possible to miss them.